### ISSUE TITLE
这个代码有问题？会报告段错误.

### ISSUE THREAD

**Issue by wkexinw** [yyzybb537/libgo#176]
`#include <iostream>
#include <libgo/coroutine.h>
using namespace std;

co_chan<int> ch_0;

void test(){
	int r=0;
	for(;;){
		ch_0 >> r;
	}
}

void test2(){
	int i=0;
	ch_0 << i;
	i++;
}

int main(int argc, char** argv)
{
	co_chan<int> ch_1;

	go [=]{
		ch_1 <<1;
	};

	go [=]{
		int r=0;
		ch_1 >> r;

		printf("%d\n",r);
	};
	//go test2;
	//go test;
	co_sched.Start();

	thread([]{ co_sleep(2000); co_sched.Stop(); }).detach();
	return 0;
}`
compile with : g++ -std=c++11 test.cpp -I /opt/libgo/include/ -L /opt/libgo/lib -llibgo -lpthread

错误：
(gdb) r
Starting program: /home/goemon/gat-1400/src/a.out 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff698f700 (LWP 8688)]

Thread 1 "a.out" received signal SIGSEGV, Segmentation fault.
0x00007ffff7ba6276 in jump_fcontext () from /home/goemon/gat-1400/lib/liblibgo.so

**Comment by yyzybb537** [comment]
rm /home/goemon/gat-1400/lib/liblibgo.so
旧版的so删了就好了
